packages/vite/src/node/plugins/asset.ts

```diff
@@ -363,9 +364,20 @@ async function fileToBuiltUrl(
       )
     }
 
-    const mimeType = mrmime.lookup(file) ?? 'application/octet-stream'
-    // base64 inlined as a string
-    url = `data:${mimeType};base64,${content.toString('base64')}`
+    if (file.endsWith('.svg')) {
+      const stringContent = content.toString()
+      // If the SVG contains some text, any transformation is unsafe, and given that double quotes would then
+      // need to be escaped, the gain to use a data URI would be ridiculous if not negative
+      if (stringContent.includes('<text')) {
+        url = `data:image/svg+xml;base64,${content.toString('base64')}`
+      } else {
+        url = svgToDataURL(stringContent)
+      }
```

Comment:
sorry for the nit again 😅 I think putting the base64 part inside `svgToDataUrl` would be nicer as the base64 string is also a data uri. And maybe `svgToDataURI` would be more accurate.